modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/MinimalApplicationModelGenerator.java

```diff
@@ -45,14 +43,45 @@ public MinimalApplicationModelGenerator(ConfigurationDependencyResolver dependen
   }
 
   /**
-   * Resolves the minimal set of {@link ComponentModel}s for a component within a flow.
+   * Resolves the minimal set of {@link ComponentModel componentModels} for the components that pass the {@link LazyComponentInitializer.ComponentLocationFilter}.
+   *
+   * @param filter to select the {@link ComponentModel componentModels} to be enabled.
+   * @return the generated {@link ApplicationModel} with the minimal set of {@link ComponentModel}s required.
+   */
+  public ApplicationModel getMinimalModel(LazyComponentInitializer.ComponentLocationFilter filter) {
+    List<ComponentModel> required = dependencyResolver.findRequiredComponentModels(filter);
+    required.stream().forEach(componentModel -> {
+      final DefaultComponentLocation componentLocation = componentModel.getComponentLocation();
+      if (componentLocation != null) {
+        final String location = componentLocation.getLocation();
+        //TODO (gfernandes) munit components has no location :S
```

Comment:
We need to review ComponentLocatorVisitor then

Reply:
Yes, we will fix this as part of DataSense tests.